krux_cloud_health/cli.py

```diff
@@ -66,29 +62,30 @@ def add_cli_arguments(self, parser):
         group.add_argument(
             '--interval',
             type=str,
-            choices=INTERVAL,
-            default='daily',
+            choices=[i.name for i in Interval],
+            default=Interval.daily.name,
             help="Set time interval of data (default: %(default)s)",
         )
 
         group.add_argument(
             '--set-date',
             type=str,
             default=None,
-            help="Retrieve cost history data for specific day ('YYYY-MM-DD') or month ('YYYY-MM'), depending on interval.",
+            help="Retrieve cost history data for specific date, depending on interval. (ex: 'YYYY-MM-DD' for daily)",
         )
 
     def run(self):
         try:
-            if self.interval == 'daily':
-                cost_history = self.cloud_health.cost_history_day(self.date_input)
-            elif self.interval == 'monthly':
-                cost_history = self.cloud_health.cost_history_month(self.date_input)
-        except ValueError as e:
+            cost_history = self.cloud_health.cost_history(self.interval)
+            if self.date_input is None:
+                cost_data = cost_history.pop()
+                self.date_input = cost_data.keys()[0]
+            else:
+                cost_data = [cost_history[i] for i in range(len(cost_history)) if cost_history[i].keys()[0] == self.date_input][0]
```

Comment:
Correct me if I am wrong.
`cost_history` is an array of dicts.

Why are we interested in popping the first element?

Can we not use the following?
cost_data = [cost_history[i] for i in range(len(cost_history)) if self.date_input in cost_history[i]][0]